torch/utils/_pytree.py

```diff
@@ -400,18 +512,21 @@ def _deque_unflatten(values: Iterable[Any], context: Context) -> Deque[Any]:
     _tuple_flatten,
     _tuple_unflatten,
     serialized_type_name="builtins.tuple",
+    flatten_with_keys_fn=_tuple_flatten_with_keys,
```

Comment:
The JAX API for registering pytrees with keypath API is different. There's a register_pytree_node and a separate register_pytree_node_with_keys. The reasoning behind this is that the flatten_with_keys_fn can be used in place of the flatten_fn. Is there a reason why we chose to diverge?

Reply:
I diverged because we already have additional registration functions on `register_pytree_node` (the serialization stuff). Also, a separate registration call seems like it would make the API will be less discoverable, so people might forget to define a `flatten_with_keys_fn`—an issue that I have been bit with by with the `tree_flatten_spec` APIs which also have a separate registration API that no one uses lol.

The ability to re-use flatten_with_keys to implement flatten is nice though, so happy to change if you think it's worth it.